manager/manager.go

```diff
@@ -49,6 +49,9 @@ type Config struct {
 	// ProtoAddr fields will be used to create listeners otherwise.
 	ProtoListener map[string]net.Listener
 
+	// AdvertiseAddr is a map of addresses to advertise, by protocol.
+	AdvertiseAddr map[string]string
```

Comment:
That model seems problematic, as well. What if one wants to create multiple listeners per protocol?

Do we envision making this work for unix sockets?